### PR TITLE
Refactor: move reactExtension into distinct files

### PR DESCRIPTION
**Pull request by rickisXP**
Follow up to #19 from https://github.com/iFixit/checkout-ui-extensions/pull/19#pullrequestreview-1607666201

Shopify seems to encourage `reactExtension(...)` being called in the same file that the component calls.

Summary/CR
--
- Created new files for each extension target that we use, and imported components to use for each of those extensions

I did this as opposed to putting the `reactExtension(...)` in the components because some components show up on both `Thank you` and `Order status` pages. This does mean that when testing components, they will have to be tested separately, since the dev console will let you choose which extension target you want to test.

QA
--
For each target, check that the respective components show up. That is:
- `purchase.checkout.block.render` shows Codice Fiscale and Purchase Order input field (when Purchase Order is selected as payment method)
- `purchase.thank-you.customer-information.render-after` shows what was input into the purchase order field at the `Thank you` page
- `customer-account.order-status.customer-information.render-after` shows the same as above in the `Order status` page
- `customer-account.order-status.block.render` shows the community link on the `Order status` page
- `purchase.thank-you.block.render` shows the same as above on the `Thank you` page

cc: @masonmcelvain

### CODE REVIEW

File: extensions/ui-extension/src/CheckoutBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { reactExtension } from '@shopify/ui-extensions-react/checkout';
+import { PurchaseOrderInput } from './fields/PurchaseOrder';
+import { CodiceFiscale } from './fields/CodiceFiscale';
+
+export default reactExtension('purchase.checkout.block.render', () => (
+   <>
+      <PurchaseOrderInput /> <CodiceFiscale />
+   </>
+));
```
Comment (masonmcelvain) [inline extensions/ui-extension/src/CheckoutBlock.tsx:9]: Hm I'm worried that this won't let us render these components in independent places. We might have to split them up like:

```suggestion
export reactExtension('purchase.checkout.block.render',
   () => <PurchaseOrderInput />
);
export reactExtension('purchase.checkout.block.render',
   () => <CodiceFiscale />
);
```

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 I'm not sure this is better or worse from a boilerplate perspective, but it seems in line with Shopify's example, nice job!

File: extensions/ui-extension/src/CheckoutBlock.tsx (hunk 1 above)
Comment (masonmcelvain) [reply extensions/ui-extension/src/CheckoutBlock.tsx:9]: It's also probably not worth changing right now. I don't think we can test multiple dynamic zones at once, and it's not clear whether we can put more than one extension in a single dynamic zone (my hunch is no)

Comment (rickisXP) [reply extensions/ui-extension/src/CheckoutBlock.tsx:9]: > It's also probably not worth changing right now. I don't think we can test multiple dynamic zones at once, and it's not clear whether we can put more than one extension in a single dynamic zone (my hunch is no)

I'm okay with changing it in a follow-up pull when we need them rendered independently. :+1:

**Comment by masonmcelvain** [comment]
@rickisXP if you've tested this locally, we can call that QA. I think the QA team has a lot on their plate right now

**Comment by rickisXP** [comment]
> @rickisXP if you've tested this locally, we can call that QA. I think the QA team has a lot on their plate right now

I did not test these changes locally. @jordycosta lmk if you need any help testing :+1:

**Comment by jordycosta** [comment]
`purchase.checkout.block.render shows Codice Fiscale and Purchase Order input field for Purchase Orders`

firefox_P6GU0DWuHD

`purchase.thank-you.customer-information.render-after shows what was input into the purchase order field at the Thank you page`

firefox_fouKnKijTQ

**Comment by jordycosta** [comment]
I was unable to get to the Order Status page using the extension targets so I couldn't confirm 3 and 4

---

In addition, I wasn't able to see the Community link on the thank you page of **purchase.thank-you-block.render**

**Comment by rickisXP** [comment]
> In addition, I wasn't able to see the Community link on the thank you page of **purchase.thank-you-block.render**

I tested this and was able to get this to show
image

> I was unable to get to the Order Status page using the extension targets so I couldn't confirm 3 and 4

I also was able to see the community component in order status
image

I'm going to qa :+1:, thanks @jordycosta!

**Pull request merged**